### PR TITLE
Add storage for profile pictures

### LINKED ISSUE #128: Add profile picture to file storage
Add profile picture table to file storage in order for all users to download profile pictures of others.

### PR DESCRIPTION
**Pull request by ahlombardini**
## Summary

  This PR adds support for user profile photos in Firebase Storage with appropriate security rules and comprehensive testing infrastructure.

  ## Changes

  ### Storage Path Management
  - **Added `profilePhotoPath()` helper** in `StoragePaths.kt` to generate paths for profile photos
    - Profile photos are stored at `profilePhotos/{userId}.{extension}`
    - The filename structure enforces the security rule that only the owner can write to their profile photo
  - **Added unit tests** for the new path helper covering various file extensions and edge cases

  ### Security Rules
  - **Profile photos have authenticated read access** - any logged-in user can view profile photos
  - **Owner-only write/delete access** - users can only upload/delete their own profile photo
  - **Filename validation** - the storage path must match the authenticated user's UID using regex pattern matching

  ### Testing Infrastructure
  - **Added `uploadFileDirect()` helper** in `FirebaseEmulator.kt`
    - Simulates admin-like file uploads by bypassing Firebase Authentication rules
    - Uses Firebase Storage Emulator REST API with `Authorization: Firebase owner` header
    - Enables testing of cross-user read access scenarios
    - Properly handles multipart upload response and extracts download tokens

  ### Test
  Added 4 instrumented tests for profile photo operations:
  1. **Owner can upload** - verifies users can upload their own profile photo with correct MIME type
  2. **Cannot upload to other users' profiles** - validates security rules prevent unauthorized writes
  3. **Owner can delete** - confirms users can remove their own profile photo
  4. **Authenticated read access** - tests that any authenticated user can read another user's profile photo by uploading a photo as a different user (via direct upload) and verifying the current user can
  access it

  ## Why These Changes

  Profile photos are a core feature for user identification and collaboration in the app. The security model balances privacy (only owner can change) with discoverability (all authenticated users can view),
  which is appropriate for a collaborative project management application where users need to recognize their teammates.

  The `uploadFileDirect()` helper enables proper testing of cross-user scenarios without requiring complex multi-user authentication flows, making the tests more maintainable and reliable.
  
  
  Summary message co-authored by @claude 
  
  Closes #128

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -13,3 +13,6 @@ local.properties
 
 # Firebase config
 app/google-services.json
+
+# Firebase rules (deployed via CI)
+firebase/**/*.rules
```
Comment (ArluAe) [inline .gitignore]: Very nice addition to keep the repo clean
